zabbixsender/zabbixsvc/zabbixsvc.go

```diff
@@ -103,9 +151,38 @@ func (h *JSONHandler) zabbixSend(metrics []*zabbixsnd.Metric) (*ZabbixResponse,
 	}
 
 	infoSplit := strings.Split(zres.Info, " ")
-	if strings.Trim(infoSplit[3], ";") != "0" || zres.Response != "success" {
-		return nil, errors.New("failed to fulfill the requests: " + strings.Trim(infoSplit[5], ";"))
+
+	failed, err := strconv.Atoi(strings.Trim(infoSplit[3], ";"))
+	if err != nil {
```

Comment:
Why are the complicating things here, rollback the code and instead where we check if send doesn't succeed increment `alerts_errors_total` there.